### PR TITLE
Add OCI rules

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

**Comment by mattmoor** [comment]
This is based on https://github.com/bazelbuild/rules_docker/pull/164

**Comment by mattmoor** [comment]
Freshly rebased and RFAL.

File: docker/contrib/push-all.bzl [hunk 1]
```diff
@@ -117,3 +125,25 @@ docker_push = rule(
     executable = True,
     implementation = _impl,
 )
+
+"""Pushes a bundle of container images.
+
+Args:
+  name: name of the rule
```
Comment (jonjohnsonjr) [inline docker/contrib/push-all.bzl]: Nit: these aren't consistent in capitalization and punctuation.

File: docker/pull.bzl [hunk 2]
```diff
@@ -90,17 +90,21 @@ _docker_pull = repository_rule(
...
-    digest: (optional) the digest of the image to pull.
-  """
-  _docker_pull(**kwargs)
+"""Pulls a container image.
+
+This rule pulls a container image into our intermediate format.  The
+output of this rule can be used interchangeably with `docker_build`.
+
+Args:
+  name: name of the rule
```
Comment (jonjohnsonjr) [inline docker/pull.bzl]: Nit: missing "." at the end

**Comment by mattmoor** [comment]
test this please

**Pull request merged**